.github/workflows/ci-asp-app.yaml

```diff
@@ -0,0 +1,133 @@
+
+name: CI-ASP
+
+# ----------------------------------------------------------------------------------------------------------
+# CI Pipeline for building ASP app and deplyoing it to Azure App Service
+# - get config file via Rattus https://github.com/Saritasa/rattus
+# - deploy to Azure App Service
+# 
+# Original Author: Rostislav Udaltsov
+# Created: 05/11/2023
+# ----------------------------------------------------------------------------------------------------------
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        required: true
+        type: string
+      BUILD_PATH:
+        required: true
+        type: string
+      TEMPLATE_PATH:
+        required: true
+        type: string
+      BUILD_FOLDER:
+        required: true
+        type: string
+      PATH_TO_SOLVE_FILE:
+        required: true
+        type: string
+      CSPROJ_FILE:
+        required: true
+        type: string
+      APP_NAME:
+        required: true
+        type: string
+      AZURE_VAULT:
+        required: true
+        type: string
+    secrets:
+      AZURE_CLIENT_ID:
+        required: true
+      AZURE_CLIENT_SECRET:
+        required: true
+      AZURE_TENANT_ID:
+        required: true
+      publish-profile:
+        required: true
+        
+jobs:
+  
+  # Installing dependencies and building the application
+  build:
+    runs-on: windows-latest
```

Comment:
that should be parametersized too from the workflow.

Reply:
done